File: swift/day8_part1_2017.swift

import Foundation

// Step 1: Read Input
let fileURL = URL(fileURLWithPath: "input.txt")
let input = try String(contentsOf: fileURL)
let lines = input.components(separatedBy: .newlines)

// Step 2: Initialize Registers
var registers: [String: Int] = [:]

// Step 3: Process Instructions
for line in lines {
    let parts = line.components(separatedBy: " ")
    let reg = parts[0]
    let op = parts[1]
    let amount = Int(parts[2])!
    let condReg = parts[4]
    let condOp = parts[5]
    let condVal = Int(parts[6])!

    // Check condition
    var cond = false
    switch condOp {
    case ">":
        cond = registers[condReg, default: 0] > condVal
    case ">=":
        cond = registers[condReg, default: 0] >= condVal
    case "<":
        cond = registers[condReg, default: 0] < condVal
    case "<=":
        cond = registers[condReg, default: 0] <= condVal
    case "==":
        cond = registers[condReg, default: 0] == condVal
    case "!=":
        cond = registers[condReg, default: 0] != condVal
    default:
        print("Invalid condition operator")
    }

    if cond {
        switch op {
        case "inc":
            registers[reg, default: 0] += amount
        case "dec":
            registers[reg, default: 0] -= amount
        default:
            print("Invalid operation")
        }
    }
}

// Step 4: Find Max Value
let maxValue = registers.values.max() ?? 0
print(maxValue)

File: javascript/day8_part1_2017.js
const fs = require('fs');

// Step 1: Read Input
const input = fs.readFileSync('input.txt', 'utf8').trim().split('\n');

// Step 2: Initialize Registers
const registers = {};

// Step 3: Process Instructions
for (const line of input) {
  const parts = line.split(' ');
  const reg = parts[0];
  const op = parts[1];
  const amount = parseInt(parts[2]);
  const condReg = parts[4];
  const condOp = parts[5];
  const condVal = parseInt(parts[6]);

  // Check condition
  let cond = false;
  switch (condOp) {
    case '>':
      cond = (registers[condReg] || 0) > condVal;
      break;
    case '>=':
      cond = (registers[condReg] || 0) >= condVal;
      break;
    case '<':
      cond = (registers[condReg] || 0) < condVal;
      break;
    case '<=':
      cond = (registers[condReg] || 0) <= condVal;
      break;
    case '==':
      cond = (registers[condReg] || 0) === condVal;
      break;
    case '!=':
      cond = (registers[condReg] || 0) !== condVal;
      break;
  }

  if (cond) {
    switch (op) {
      case 'inc':
        registers[reg] = (registers[reg] || 0) + amount;
        break;
      case 'dec':
        registers[reg] = (registers[reg] || 0) - amount;
        break;
    }
  }
}

// Step 4: Find Max Value
let maxValue = 0;
for (const value of Object.values(registers)) {
  if (value > maxValue) {
    maxValue = value;
  }
}

console.log(maxValue);